crates/runtime/src/cluster/mod.rs

```diff
@@ -94,7 +121,24 @@ pub async fn initialize_cluster_executor(
                 .into(),
             })?;
 
-    let scheduler = SchedulerGrpcClient::new(scheduler_connection)
+    let Some(api_key) = rt.config.cluster.cluster_api_key.clone() else {
+        return Err(FailedToStartClusterExecutor {
+            source: "Unable to start executor without an API key".into(),
+        });
+    };
+
+    let interceptor = move |mut req: tonic::Request<()>| {
+        req.metadata_mut().insert(
+            "authorization",
+            format!("Bearer {api_key}")
+                .parse()
+                .map_err(|e| tonic::Status::invalid_argument("Invalid API key"))?,
+        );
+
+        Ok(req)
+    };
+
+    let scheduler = SchedulerGrpcClient::with_interceptor(scheduler_connection, interceptor)
```

Comment:
Executor consumes the client, so we decorate API key headers before passing it along